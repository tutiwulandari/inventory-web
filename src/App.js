import './App.css';
import {Route, Switch, BrowserRouter as Router} from "react-router-dom";
import routes from "./configs/routes";
import {Provider} from "react-redux";
import store  from "./configs/store"
import Nav from './components/Navbar'
import Jumbotron from './components/JumbotronComponent';

function App() {
    return (
        <Provider store={store}>
        <Router>
          <Nav/>
            <Switch>
                {routes.map((route, index) =>
                    <Route key={index} path={route.path} exact>
                        {route.component}
                    </Route>)}
            </Switch>
        </Router>
        </Provider>
    );
}

export default App;
