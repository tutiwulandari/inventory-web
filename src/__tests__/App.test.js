import App from "../App"
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'



describe('App Component', () => {

    it('display the result', function (){
        const tree = renderer.create(<App/>).toJSON()
        expect(tree).toMatchSnapshot()
    });

    it('should have one provider', function () {
        const  wrapper = shallow(<App/>)
        expect(wrapper.find('Provider')).toHaveLength(1)
    });

    it('should not have div', function () {
        const wrapper = shallow(<App/>)
        expect(wrapper.find('div')).toHaveLength(0)
    });

    it('should have two render', function (){
        const wrapper = shallow(<App/>)
        expect(wrapper.render()).toHaveLength(2)
    });

    it('should have one browser router', function (){
        const wrapper = shallow(<App/>)
        expect(wrapper.find('BrowserRouter')).toHaveLength(1)
    });

    it('should have one Switch', function (){
        const wrapper = shallow(<App/>)
        expect(wrapper.find('Switch')).toHaveLength(1)
    })
})