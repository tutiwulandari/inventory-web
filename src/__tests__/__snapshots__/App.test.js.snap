// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component display the result 1`] = `
Array [
  <div>
    <nav>
      <ul
        style={
          Object {
            "backgroundColor": "#333",
            "listStyleType": "none",
            "margin": 0,
            "overflow": "hidden",
            "padding": 0,
          }
        }
      >
        <li
          style={
            Object {
              "float": "left",
            }
          }
        >
          <a
            href="/unit"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
                "display": "block",
                "padding": "14px 16px",
                "textAlign": "center",
                "textDecoration": "none",
              }
            }
          >
            Unit Form
          </a>
        </li>
        <li
          style={
            Object {
              "float": "left",
            }
          }
        >
          <a
            href="/units"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
                "display": "block",
                "padding": "14px 16px",
                "textAlign": "center",
                "textDecoration": "none",
              }
            }
          >
            Unit List
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  <div>
    <div>
      <a
        href="/unit"
        onClick={[Function]}
      >
        <button
          aria-label={null}
          className="btn btn-info btn-md"
          onClick={[Function]}
          style={
            Object {
              "marginTop": "50px",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-plus fa-w-14 "
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={
              Object {
                "ma": "1px",
              }
            }
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Add
        </button>
      </a>
    </div>
    <div
      className="d-flex justify-content-center mb-3"
    >
      <h1
        style={
          Object {
            "fontFamily": "sans-serif",
          }
        }
      >
         List Unit 
      </h1>
    </div>
    <table
      align="center"
      className="table table-bordered table-hover"
      style={
        Object {
          "width": "75%",
        }
      }
    >
      <thead
        className="thead-dark"
      >
        <tr
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <th
            scope="row"
          >
            ID
          </th>
          <th
            scope="row"
          >
            Code
          </th>
          <th
            scope="row"
          >
            Description
          </th>
          <th
            scope="row"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;
